### ISSUE TITLE
Unable to ignore Helm chart templates

### ISSUE THREAD

**Issue by carlsmedstad** [adrienverge/yamllint#408]
Helm uses the `.yaml` extension for template files that are not valid YAML. Unfortunately I have not found a solution to have yamllint ignore them completely, since the `.gitignore` method of ignoring is lacking. For example, the following config will ignore them, but not if you're standing in the `templates/` directory:
```yaml
ignore:
  **/templates/*.yaml
```

One possible resolution to this would be to implement an `ignore_regex` keyword that would take a list of regular expression and try to match that against the full absolute path of files. I can make a PR if this is desirable.

**Comment by adrienverge** [comment]
Hello, I'm sorry but I'm not sure I have fully understood your problem. Anyway, if you're inside the `templates` directory (i.e. `$PWD` is `templates`), it's totally normal that `**/templates/*.yaml` will not exclude your `*.yaml` files. These paths are relative to the current directory.

When implementing the feature, we made a choice between regexes and `.gitignore`-style paths, the latter was chosen because it's more "newbie-friendly" and usually allows to achieve the same goal.

**Comment by carlsmedstad** [comment]
The issue is that it's impossible to identify if a file is a Helm chart with only its name. You need the directory it's located in to make a reasonable exclusion, and this is not possible with the `.gitignore` way of matching files.

My request is for users to in some way be able to ignore Helm charts since this is not an uncommon use-case. Having an alternate method of ignoring files (regular expressions) would fulfill this request. Is this something you would accept a PR for?

**Comment by adrienverge** [comment]
I'm sorry but I don't understand what is the problem you are facing. I don't know what a Helm chart is, is it relevant here? Maybe you could make your problem clear by describing it in a different way? E.g. an example of what your current directory (`$PWD`) contains, what command you use to run yamllint, what you are expecting, etc.

**Comment by carlsmedstad** [comment]
Helm is a tool used to manage Kubernetes resources. The standard project structure looks like this:
```sh
$ tree
.
├── charts
├── Chart.yaml
├── templates
│   ├── deployment.yaml
│   ├── _helpers.tpl
│   ├── hpa.yaml
│   ├── ingress.yaml
│   ├── NOTES.txt
│   ├── serviceaccount.yaml
│   ├── service.yaml
│   └── tests
│       └── test-connection.yaml
└── values.yaml
```

One unfortunate thing is that the files residing under `templates/` directory use the `.yaml` extension while not actually being syntactically correct YAML. They look like this:
```yaml
{{- if .Values.serviceAccount.create -}}
apiVersion: v1
kind: ServiceAccount
metadata:
  name: {{ include "my-service.serviceAccountName" . }}
  labels:
    {{- include "my-service.labels" . | nindent 4 }}
  {{- with .Values.serviceAccount.annotations }}
  annotations:
    {{- toYaml . | nindent 4 }}
  {{- end }}
{{- end }}
```

I would like to be able to have yamllint ignore these files. The names can be arbitrary, but the files always reside in a directory called `templates/` and use the extension `.yaml`.

**Comment by adrienverge** [comment]
Carl, I'm sorry but you're still not describing what the problem is, and you didn't answer my questions. I have limited time to dedicate to this.
Helm and Kubernetes seem irrelevant to this, aren't they? You have a bunch of Jinja2 files that end in `.yaml` that you don't want yamllint to lint, right? I just tried creating a dir `templates` and exclude it using the same `exclude` configuration as you posted, and it works as expected.

**Comment by carlsmedstad** [comment]
My bad. I just realized why I'm experiencing issues. I use a Vim plugin for running yamllint and that invoked it on temporary files with a completely different directory structure: `['/bin/bash', '-c', '''yamllint'' -f parsable ''/tmp/nvimSvMPRC/2/ingress.yaml''']`. This explains why the pattern does not work for me.

Terribly sorry for taking up your time!
